pkg/queue/readiness/probe.go

```diff
@@ -56,6 +85,36 @@ func (p *Probe) IsAggressive() bool {
 
 // ProbeContainer executes the defined Probe against the user-container
 func (p *Probe) ProbeContainer() bool {
+	// https://en.wikipedia.org/wiki/Double-checked_locking is implemented below.
```

Comment:
Is double checked locking needed here? We're not optimizing this for microsecond performance I think? Just usual locking might be better to keep the code a little bit more straightforward.

Reply:
Done. Simplified a lot without 2 step locking